File: RecyclerMain/java/com/example/administrator/recyclerview/main/MainActivity.java
package com.example.administrator.recyclerview.main;

import android.content.Intent;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.administrator.recyclerview.R;
import com.example.administrator.recyclerview.main_second.SecondActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.mainActivityViewPager)
    ViewPager viewPager;
    @BindView(R.id.mainActivityTabLayout)
    TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setViewPager();
    }

    private void setViewPager() {
        MainViewPagerAdapter mainViewPagerAdapter = new MainViewPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(mainViewPagerAdapter);
        tabLayout.setupWithViewPager(viewPager);
    }

    @OnClick(R.id.mainActivityButton)
    public void onClickStart() {
        Intent intent = new Intent(MainActivity.this, SecondActivity.class);
        startActivity(intent);
    }
}

File: RecyclerMain_Kotlin/kotlin/com/example/administrator/recyclerview/main/main/MainActivity.kt
package com.example.administrator.recyclerview.main.main

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

import com.example.administrator.recyclerview.R
import com.example.administrator.recyclerview.main_second.SecondActivity

import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setViewPager()
    }

    private fun setViewPager() {
        val mainViewPagerAdapter = MainViewPagerAdapter(supportFragmentManager)
        mainActivityViewPager.adapter = mainViewPagerAdapter
        mainActivityTabLayout.setupWithViewPager(mainActivityViewPager)

        mainActivityButton.setOnClickListener {
            startActivity<SecondActivity>()
        }
    }
}
